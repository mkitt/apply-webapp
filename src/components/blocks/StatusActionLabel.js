// @flow
import React from 'react'
import glamorous from 'glamorous'
import { RocketIcon } from '../elements/Icons'

type Props = {
  children: React.Element<*>,
}

export default ({ children, ...props }: Props) => (
  <View {...props}>
    <RocketIcon width={24} height={24} marginRight={12} />
    <Text>{children}</Text>
  </View>
)

// -------------------------------------

const rules = {
  display: 'flex',
  flexDirection: 'row',
}

const View = glamorous.div(
  rules,
  ({ children, ...props }) => ({ ...props }),
)

const Text = glamorous.span({
  fontSize: 14,
  fontWeight: 600,
  textTransform: 'uppercase',
  color: '#666',
})
