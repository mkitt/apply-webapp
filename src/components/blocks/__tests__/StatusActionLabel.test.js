import React from 'react'
import renderer from 'react-test-renderer'
import StatusActionLabel from '../StatusActionLabel'

describe('<StatusActionLabel />', () => {
  it('shows the correct snapshot tree for the <StatusActionLabel> block', () => {
    const component = (
      <StatusActionLabel status="Status action label" />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <StatusActionLabel inCard> block', () => {
    const component = (
      <StatusActionLabel inCard status="Status action label in card" />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
