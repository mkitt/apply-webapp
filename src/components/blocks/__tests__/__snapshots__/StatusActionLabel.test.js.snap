// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusActionLabel /> shows the correct snapshot tree for the <StatusActionLabel inCard> block 1`] = `
.glamor-0,
[data-glamor-0] {
  margin-right: 12px;
}

.glamor-1,
[data-glamor-1] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
}

.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  align-items: center;
  color: #666;
  position: absolute;
  top: 16px;
  right: 16px;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

<div
  className="glamor-2"
>
  <svg
    className="glamor-0"
    height={24}
    viewBox="0 0 24 24"
    width={24}
  >
    <g
      fill="currentColor"
    >
      <path
        d="M24,0 C24,0 23.4836048,9.80252996 16.4820958,15.2719707 L16.4128111,19.0346251 L11.4820194,24 L11.4746834,20.2882431 L9.63105705,18.6292449 C8.29671435,22.7691304 1.26439406,22.7691304 1.26439406,22.7691304 C1.26439406,22.7691304 1.26439406,16.3396736 5.58319762,14.4450082 L3.94220876,12.6079505 L-2.84217094e-14,12.6079505 L5.07440302,7.66843342 L8.7285621,7.58627726 C14.4472436,0.671411685 24,0 24,0 Z M16.425,10.1644246 C17.8333261,10.1644246 18.975,9.01830529 18.975,7.60449544 C18.975,6.19068562 17.8333261,5.0445663 16.425,5.0445663 C15.0166739,5.0445663 13.875,6.19068562 13.875,7.60449544 C13.875,9.01830529 15.0166739,10.1644246 16.425,10.1644246 Z"
      />
    </g>
  </svg>
  <span
    className="glamor-1"
  />
</div>
`;

exports[`<StatusActionLabel /> shows the correct snapshot tree for the <StatusActionLabel> block 1`] = `
.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  align-items: center;
  color: #666;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-0,
[data-glamor-0] {
  margin-right: 12px;
}

.glamor-1,
[data-glamor-1] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
}

<div
  className="glamor-2"
>
  <svg
    className="glamor-0"
    height={24}
    viewBox="0 0 24 24"
    width={24}
  >
    <g
      fill="currentColor"
    >
      <path
        d="M24,0 C24,0 23.4836048,9.80252996 16.4820958,15.2719707 L16.4128111,19.0346251 L11.4820194,24 L11.4746834,20.2882431 L9.63105705,18.6292449 C8.29671435,22.7691304 1.26439406,22.7691304 1.26439406,22.7691304 C1.26439406,22.7691304 1.26439406,16.3396736 5.58319762,14.4450082 L3.94220876,12.6079505 L-2.84217094e-14,12.6079505 L5.07440302,7.66843342 L8.7285621,7.58627726 C14.4472436,0.671411685 24,0 24,0 Z M16.425,10.1644246 C17.8333261,10.1644246 18.975,9.01830529 18.975,7.60449544 C18.975,6.19068562 17.8333261,5.0445663 16.425,5.0445663 C15.0166739,5.0445663 13.875,6.19068562 13.875,7.60449544 C13.875,9.01830529 15.0166739,10.1644246 16.425,10.1644246 Z"
      />
    </g>
  </svg>
  <span
    className="glamor-1"
  />
</div>
`;
